assignments/lambda-classes.js

```diff
@@ -1 +1,108 @@
 // CODE here for your Lambda Classes
+
+class Person {
+    constructor(attributes){
+        this.name = attributes.name;
+        this.age = attributes.age;
+        this.location = attributes.location;
+        this.gender = attributes.gender;
+    }
+    speak() {
+        return `Hello my name is ${this.name}, I am from ${this.location}`;
+    }
+}
+
+class Instructor extends Person{
+    constructor(instructorAttributes){
+        super(instructorAttributes);
+        this.specialty = instructorAttributes.specialty;
+        this.favLanguage = instructorAttributes.favLanguage;
+        this.catchPhrase = instructorAttributes.catchPhrase;
+    }
+    demo(subject){
+        return `Today we are learning about ${subject}`
+    }
+    grade(student, subject){
+        return `${student.name} receives a perfect score on ${subject}`
+    }
+}
+
+class Student extends Person {
+    constructor(studentAttributes){
+        super(studentAttributes);
+        this.previousBackground = studentAttributes.previousBackground;
+        this.className = studentAttributes.className;
+        this.favSubjects = studentAttributes.favSubjects;
+    }
+    listsSubjects(arr) {
+        return arr.join();
```

Comment:
The requirements asked for each subject to be logged out one at a time, so a for loop would be more appropriate here. Aside from that, try adding a delimiter as an argument for `join`, like `, `:

```js
arr.join(', ');
```

That would make the the resulting string a little nicer, since it will add a `, ` between each string in the array.